### PR TITLE
SwFormControl

### PR DESCRIPTION
**Pull request by tomecko**

### CODE REVIEW

File: src/input/form-control/form-control.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+function SwFormControl(props: ISwFormControlProps) {
+  const { formatter, onChange, onParsedChange, parser, value, ...rest } = props;
+  return (
+    <FormControl
+      {...rest}
+      onChange={(e: React.FormEvent<FormControl>) => {
+        if (Boolean(onChange)) {
+          onChange(e);
+        }
+        Maybe.fromNull((e.currentTarget as any).value)
```
Comment (ulfryk) [inline src/input/form-control/form-control.tsx]: why `Maybe.fromNull` here while `if (Boolean(onChange)) ` in line 13 ?

maybe make it consistent :)

File: src/input/form-control/form-control.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          .forEach(val => {
+            if (Boolean(onParsedChange)) {
```
Comment (ulfryk) [inline src/input/form-control/form-control.tsx]: maybe `flatMap` and `fromNull` for sake of consistency?

File: src/input/form-control/form-control.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+              onParsedChange(parser(val));
```
Comment (ulfryk) [inline src/input/form-control/form-control.tsx]: why have both `onParsedChange` and `onChange` ?
shouldn't parser be applied to value passed to onChange if exists ?

Comment (tomecko) [reply src/input/form-control/form-control.tsx]: I did not attempt to modify `event` passed to `onChange`… that's the reason for `onParsedChange` existence

**Pull request merged**